### PR TITLE
feat: validate token before get from DB, use GetTokenByValueOrId instead of GetTokenByValue

### PR DESCRIPTION
**Pull request by notdu**
Optimize token validation and retrieval
- Validate the token before retrieving it from the database to reduce unnecessary database calls.
- Replace `GetTokenByValue` with `GetTokenByValueOrId` for token retrieval, as querying by token value (text) is inefficient for indexing and querying.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

File: object/token.go [hunk 1]
```diff
@@ -137,18 +159,39 @@ func GetTokenByRefreshToken(refreshToken string) (*Token, error) {
 	return &token, nil
 }
 
-func GetTokenByTokenValue(tokenValue, tokenTypeHint string) (*Token, error) {
+func GetTokenByRefreshTokenOrName(refreshToken, tokenName string) (*Token, error) {
```
Comment (dacongda) [inline object/token.go]: this function is totally  same with `GetTokenByAccessTokenOrName`, it should be remove and name as `GetTokenByTokenOrName`

**Comment by notdu** [comment]
Hi @dacongda , I’ve made the changes and force-pushed them. Could you take another look when you have a moment?

**Review by dacongda — CHANGES_REQUESTED** [review]

File: object/token.go (hunk 1 above)
Comment (dacongda) [reply object/token.go]: you can just pack the same code to a new func instead of remove them

Comment (notdu) [reply object/token.go]: hi @dacongda , i've just updated

Comment (dacongda) [reply object/token.go]: > hi @dacongda , i've just updated

I know, and those code is same with previous one,  I just suggest that make the same part between `GetTokenByRefreshTokenOrName` and GetTokenByAccessTokenOrName to be a new function

Comment (notdu) [reply object/token.go]: @dacongda, these two methods are distinct: one retrieves tokens by RefreshTokenHash, while the other retrieves them by AccessTokenHash

Comment (dacongda) [reply object/token.go]:
```go

		owner, name := util.GetOwnerAndNameFromIdNoCheck(tokenId)
		token = Token{Owner: owner, Name: name}
		existed, err = ormer.Engine.Get(&token)
		if err != nil {
			return nil, err
		}
	
```

these line can be a new function named `getTokenByTokenId`

Comment (notdu) [reply object/token.go]: hi @dacongda, i've updated

**Review by dacongda — APPROVED** [review]
lgtm

**Comment by hsluoyz** [comment]
@notdu I don't think your changes are reasonable. DB query is already indexed, already very fast, no need to improve. Even if the attacker wants to access the login page of Casdoor, it requires several DB queries like get-application. DB queries are inevitable.

And the token ID is also hard to understand, making code ugly. Don't know what you are trying to resolve.

First make issue first.

**Pull request closed without merging**